src/ConstructiveSolidGeometry/SurfacePrimitives/Rectangle.jl

```diff
@@ -0,0 +1,197 @@
+struct Rectangle{TN,T,TL,TW} <: AbstractSurfacePrimitive{T}
+    l::TL
+    w::TW
+    loc::T
+    function Rectangle( normal::Union{Val{:x}, Val{:y}, Val{:z}},
+                        ::Type{T},
+                        l::Union{T, <:AbstractInterval{T}},
+                        w::Union{T, <:AbstractInterval{T}},
+                        loc::T) where {T}
+        new{typeof(normal),T,typeof(l),typeof(w)}(l,w,loc)
+    end
+end
+
+# Convenience functions
+const RectangleX{T,TL,TW} = Rectangle{Val{:x},T,TL,TW}
+const RectangleY{T,TL,TW}  = Rectangle{Val{:y},T,TL,TW}
+const RectangleZ{T,TL,TW} = Rectangle{Val{:z},T,TL,TW}
+
+RectangleX(args...) = Rectangle(Val(:x), args...)
+RectangleY(args...)  = Rectangle(Val(:y), args...)
+RectangleZ(args...) = Rectangle(Val(:z), args...)
+
+#Constructors
+RectangleX(b::Box{T}, x::Real) where {T} = RectangleX(T, b.y, b.z, T(x))
+RectangleY(b::Box{T}, y::Real) where {T} = RectangleY(T, b.x, b.z, T(y))
+RectangleZ(b::Box{T}, z::Real) where {T} = RectangleZ(T, b.x, b.y, T(z))
+
+function Rectangle(;normal = Val(:z), lMin = -1, lMax = 1, wMin = -1, wMax = 1, loc = 0)
+    T = float(promote_type(typeof.((lMin, lMax, wMin, wMax, loc))...))
+    l = lMax == -lMin ? T(lMax) : T(lMin)..T(lMax)
+    w = wMax == -wMin ? T(wMax) : T(wMin)..T(wMax)
+    Rectangle(normal, T, l, w, T(loc))
+end
+
+Rectangle(normal, lMin, lMax, wMin, wMax, loc) = Rectangle(; normal = normal, lMin = lMin, lMax = lMax, wMin = wMin, wMax = wMax, loc = loc)
+
+function Rectangle(normal::Union{Val{:x}, Val{:y}, Val{:z}}, l::L, w::W, loc::C) where {L<:Real, W<:Real, C<:Real}
+    T = float(promote_type(L,W,C))
+    Rectangle(normal, T, T(l/2), T(w/2), T(loc))
+end
+
+@inline in(p::AbstractCoordinatePoint, r::RectangleX) = begin
+    _in_y(p, r.l) && _in_z(p, r.w) && _isapprox_x(p, r.loc)
+end
+
+@inline in(p::AbstractCoordinatePoint, r::RectangleY) = begin
+    _in_x(p, r.l) && _in_z(p, r.w) && _isapprox_y(p, r.loc)
+end
+
+@inline in(p::AbstractCoordinatePoint, r::RectangleZ) = begin
+    _in_x(p, r.l) && _in_y(p, r.w) && _eq_z(p, r.loc)
+end    
```

Comment:
Is there a reason why for `x` and `y` we use `_is_approx` while for `z` we use `_eq`?

Reply:
I did not consider rotations. I will add `_is_approx_z`